scan_test.go

```diff
@@ -121,6 +121,59 @@ func TestZoneParserIncludeDisallowed(t *testing.T) {
 	}
 }
 
+func TestZoneParserAddressAAAA(t *testing.T) {
+	zone := `
+1.example.org. 600 IN AAAA ::1
+2.example.org. 600 IN AAAA ::FFFF:127.0.0.1`
+
+	wantRRs := []*AAAA{
+		&AAAA{Hdr: RR_Header{Name: "1.example.org."}, AAAA: net.IPv6loopback},
+		&AAAA{Hdr: RR_Header{Name: "2.example.org."}, AAAA: net.ParseIP("::FFFF:127.0.0.1")},
+	}
+
+	wantIdx := 0
+	zp := NewZoneParser(strings.NewReader(zone), "", "")
```

Comment:
NewRR used the same code; these two tests can be collapsed to just checking the new code works, by testing
~~~
1.example.org. 600 IN AAAA ::1
2.example.org. 600 IN AAAA ::FFFF:127.0.0.1`
"1.bad.example.org. 600 IN A ::1",
"2.bad.example.org. 600 IN A ::FFFF:127.0.0.1",
"3.bad.example.org. 600 IN AAAA 127.0.0.1",
~~~

Reply:
Done.